gradio/components.py

```diff
@@ -898,13 +1023,40 @@ def __init__(
         self.type = type
         self.test_input = self.choices
         self.interpret_by_tokens = False
-        IOComponent.__init__(self, label=label, css=css, **kwargs)
+        IOComponent.__init__(
+            self,
+            label=label,
+            show_label=show_label,
+            interactive=interactive,
+            css=css,
+            visible=visible,
+            **kwargs,
+        )
 
-    def get_template_context(self):
+    def get_config(self):
         return {
             "choices": self.choices,
             "default_value": self.default_value,
-            **IOComponent.get_template_context(self),
+            **IOComponent.get_config(self),
+        }
+
+    @staticmethod
+    def update(
+        value: Optional[Any] = None,
```

Comment:
I assume `value` updates `default_selected` but this is not clear. I think we should just change the name of the first parameter in every component to be `value`